utils/release.bash

```diff
@@ -237,10 +239,14 @@ function guessGitBranchInformation(){
   #BRANCH_NAME="stable-2.235"
   #BRANCH_NAME="master"
 
+  ## If needed, set BRANCH_NAME
+  if [ -z "$BRANCH_NAME" ];then
+    DEFAULT_BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD)"
+    : "${BRANCH_NAME:=$DEFAULT_BRANCH_NAME}"
+  fi
```

Comment:
```suggestion
DEFAULT_BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD)"
: "${BRANCH_NAME:=$DEFAULT_BRANCH_NAME}"
```

Otherwise the code will fail with the error `./release.bash: line 243: BRANCH_NAME: unbound variable`
due to the `set -u` option enabled.

Reply:
That's a fallback that I want to avoid because I don't want it to fail if BRANCH_NAME is set but git is not available

Reply:
```suggestion
    DEFAULT_BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD || echo "master")"
    : "${BRANCH_NAME:=$DEFAULT_BRANCH_NAME}"
```

Reply:
```suggestion
  DEFAULT_BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD || echo 'master')"
  : "${BRANCH_NAME:=$DEFAULT_BRANCH_NAME}"
```

Reply:
As explained on IRC, the purpose is to be able to run the command make test to run some bats test from a container which doesn't have git installed